src/plugins/intel_gpu/src/graph/graph_optimizer/select_preferred_formats.cpp

```diff
@@ -37,9 +37,11 @@ void select_preferred_formats::run(program& p) {
         try {
             dnnl::primitive_desc prim_desc;
             if (n->is_type<convolution>()) {
+                n->get_output_layouts(true); // initialize output layouts
```

Comment:
why do we need it?

Reply:
`auto desc = onednn::get_convolution_descriptor(*n->get_kernel_impl_params(), dnnl::memory::format_tag::any);`
Without it, program_node::output_layouts and program_node::valid_output_layouts will be filled with format::bfyx and false.
We are creating conv descriptor with n->get_kernel_impl_params(),
```
virtual std::unique_ptr<kernel_impl_params> get_kernel_impl_params() const {
    return get_kernel_impl_params(get_input_layouts(), output_layouts);
}
```
However, output_layouts were not initialized in this test. Therefore, we should explicitly call n->get_output_layouts(true) to initialize its output_layouts with proper values.
Fixing program_node behavior would be more elegant, but it's too big deal compared to fixing just one unit test.